internal/client/client.go

```diff
@@ -175,6 +177,33 @@ func (c *CSAPI) SyncUntil(t *testing.T, since, key string, check func(gjson.Resu
 	}
 }
 
+//RegisterUser will register the user with given parameters and
+// return user ID & access token, and fail the test on network error
+func (c *CSAPI) RegisterUser(t *testing.T, localpart, password string) (userID, accessToken string) {
+	t.Helper()
+	reqBody := map[string]interface{}{
+		"auth": map[string]string{
+			"type": "m.login.dummy",
+		},
+		"username": localpart,
+		"password": password,
+	}
+	res := c.MustDo(t, "POST", []string{"_matrix", "client", "r0", "register"}, reqBody)
+	must.MatchResponse(t, res, match.HTTPResponse{
```

Comment:
This is a helper function. You don't need to do assertions here (stuff in the `must` or `match` packages). Note that in all the other functions here we don't do this either. This is because these helper functions are not the main point of the test, and are just a bit of code to help set up the test. It's important to make it clear how the test is set up, and having assertions here can make things harder to read.